alerting/src/main/kotlin/org/opensearch/alerting/alerts/AlertIndices.kt

```diff
@@ -331,26 +358,48 @@ class AlertIndices(
                 indicesToDelete.add(indexMetaData.index.name)
             }
         }
+        return indicesToDelete
+    }
 
+    private fun deleteAllOldHistoryIndices(indicesToDelete: List<String>) {
         if (indicesToDelete.isNotEmpty()) {
             val deleteIndexRequest = DeleteIndexRequest(*indicesToDelete.toTypedArray())
-            val deleteIndexResponse = client.admin().indices().delete(deleteIndexRequest).actionGet()
-
-            if (!deleteIndexResponse.isAcknowledged) {
-                logger.error("Could not delete one or more Alerting history indices: $indicesToDelete. Retrying one by one.")
-                for (index in indicesToDelete) {
-                    try {
-                        val singleDeleteRequest = DeleteIndexRequest(*indicesToDelete.toTypedArray())
-                        val singleDeleteResponse = client.admin().indices().delete(singleDeleteRequest).actionGet()
+            client.admin().indices().delete(
+                deleteIndexRequest,
+                object : ActionListener<AcknowledgedResponse> {
+                    override fun onResponse(deleteIndicesResponse: AcknowledgedResponse) {
+                        if (!deleteIndicesResponse.isAcknowledged) {
+                            logger.error("Could not delete one or more Alerting history indices: $indicesToDelete. Retrying one by one.")
+                            deleteOldHistoryIndex(indicesToDelete)
+                        }
+                    }
+                    override fun onFailure(e: Exception) {
+                        logger.error("Delete for Alerting History Indices $indicesToDelete Failed. Retrying one By one.")
+                        deleteOldHistoryIndex(indicesToDelete)
+                    }
+                }
+            )
+        }
+    }
 
-                        if (!singleDeleteResponse.isAcknowledged) {
-                            logger.error("Could not delete one or more Alerting history indices: $index")
+    private fun deleteOldHistoryIndex(indicesToDelete: List<String>) {
+        for (index in indicesToDelete) {
+            val singleDeleteRequest = DeleteIndexRequest(*indicesToDelete.toTypedArray())
+            client.admin().indices().delete(
+                singleDeleteRequest,
+                object : ActionListener<AcknowledgedResponse> {
+                    override fun onResponse(acknowledgedResponse: AcknowledgedResponse?) {
+                        if (acknowledgedResponse != null) {
+                            if (!acknowledgedResponse.isAcknowledged) {
+                                logger.error("Could not delete one or more Alerting history indices: $index")
+                            }
                         }
-                    } catch (e: IndexNotFoundException) {
+                    }
+                    override fun onFailure(e: Exception) {
                         logger.debug("$index was already deleted. ${e.message}")
```

Comment:
Do you want to wrap this in a if IndexNotFoundException check? Otherwise seems we say it was already deleted on every exception.

Reply:
I too thought about it but then on a delete index request we don't anticipate any other exception apart from index deleted exception. If the index is not deleted we get the response as `Acknowledged False`.

Reply:
Hmm but what about network exceptions and timeouts and others?

Reply:
Right we can modify the log message keeping the exception same?